### PR TITLE
Multi-broker, multi-partition Kafka test #5968

### PR DESCRIPTION
**Pull request by philip-stoev**
Implement a mzcompose based multi-broker, multi-partition test.

This required the following changes:
- remove the use of multiple brokers from the kafka-ssl test, as that test
  only used multiple brokers in order to provide some multi-broker coverage

- add 'sorted-messages' option to $ kafka-verify .This way multi-partition sinks can be examined with kafka-verify 

- add 'signal' to the kill-services mzcompose step

- add 'replication_factor' to $ kafka-create-topic, make the default -1

---

The test uses liberal sleeps in order to not deal with any kafka/zookeeper oddities. Therefore it is relegated to the nightly build.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/6259)

### CODE REVIEW

File: src/testdrive/src/action/kafka/create_topic.rs [hunk 1]
```diff
@@ -150,14 +157,18 @@ impl Action for CreateTopicAction {
         );
         let partitions = i32::try_from(self.partitions)
             .map_err(|_| format!("partition count must fit in an i32: {}", self.partitions))?;
-        let new_topic = NewTopic::new(&topic_name, partitions, TopicReplication::Fixed(1))
```
Comment (philip-stoev) [inline src/testdrive/src/action/kafka/create_topic.rs:153]: It took me forever to find this default that sets the replication factor to 1 .

**Review by elindsey — APPROVED** [review]
This looks good to me!

**Comment by philip-stoev** [comment]
@benesch can you please take a look at the testdrive-related stuff in this PR. I do not feel comfortable merging testdrive changes without your review. Thanks!

**Review by benesch — APPROVED** [review]
Oh, sorry! @elindsey's +1 is good enough for me, but this also LGTM.

**Comment by philip-stoev** [comment]
This PR got borked from the rebase. Will open a new one just for the purpose of merging. Thanks!

**Pull request closed without merging**